### ISSUE TITLE
Support for Nesting of Blade Text properties

### ISSUE THREAD

**Issue by pritishvaidya** [razorpay/blade#1177]
Context: Nested Text in Blade for any composite component doesn't support inheritance for common theme props such as `color, fontWeight` etc

Example (from blade-old):
```
    <Flex flexDirection="row" alignItems="center">
      <View>
       ...
        <Text color="light.960" weight="regular" size="medium">
          count
        </Text>
        <Text color="light.980" weight="regular" size="medium">
          name... approved the payout
        </Text>
      </View>
    </Flex>
```

Expected
```
  <Text color="light.960" weight="regular" size="medium">
    count <Text color="light.980">name... approved the payout</Text>
  </Text>
```

[Support PR](https://github.com/razorpay/blade/pull/1170) (for React Native)

**Comment by divyanshu013** [comment]
Adding the MoMs here from Slack:

MoM:
- tl;dr / conclusion - wait to discover more usecases and patterns where passing props into nested typography based components can simplify the code.

Discussed:
- Benefits: can help in authoring code with deeply nested typography components
- Open [issue](https://github.com/facebook/react-native/issues/31955) on RN which will likely mess up with alignments (we’ll have to wait for this to be closed by RN team)
- Say we introduce this, it can lead to confusions with usage in cases such as:

```jsx
<Text size="large">
  I am 
  <Text>large</Text> <-- default size is medium but would render as large
</Text>
```

- Another caveat is right now the props themselves aren’t exactly the same in terms of actual computed style values. Eg - medium of text might not be same as medium of code / amount / etc.
  - Some components don’t have all the props (eg. link with xsmall)

**Comment by divyanshu013** [comment]
Related #1094 (need to also think about streamlining `size` prop first)
